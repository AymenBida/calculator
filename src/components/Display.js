import PropTypes from 'prop-types';
import './display.css';

const TriggerLittle = (result) => {
  if (result.length < 11) return 'display';

  return 'display display--little';
};

const Display = ({ result }) => (
  <div className={TriggerLittle(result)}>{ result }</div>
);

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
