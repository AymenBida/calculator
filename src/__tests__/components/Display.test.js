import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../../components/Display';

it('renders correctly with "0" per default', () => {
  const tree = renderer.create(<Display />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with "result" if given', () => {
  const tree = renderer.create(<Display result="177" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with a special class if the result is longer than 11 characters', () => {
  const tree = renderer.create(<Display result="17716251456985" />).toJSON();
  expect(tree).toMatchSnapshot();
});
