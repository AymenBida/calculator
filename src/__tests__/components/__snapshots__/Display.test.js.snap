// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with "0" per default 1`] = `
<div
  className="display"
>
  0
</div>
`;

exports[`renders correctly with "result" if given 1`] = `
<div
  className="display"
>
  177
</div>
`;

exports[`renders correctly with a special class if the result is longer than 11 characters 1`] = `
<div
  className="display display--little"
>
  17716251456985
</div>
`;
